### PR TITLE
refactor(ci): verification workflow

### PR DESCRIPTION
**Pull request by chrisgacsal**
## Description

* split verification into multiple jobs running parallel
* add caching for `licensei`
* add caching for `golangci-lint`
* add caching of Go dependencies and builds

## Type of Change

[ ] Bug Fix  
[x] New Feature  
[ ] Breaking Change  
[x] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

**Comment by chrisgacsal** [comment]
The run time for the Verification has been reduced from ~14-18 mins to ~2-3mins for subsequent CI runs. Merging this PR will ensure that the `main` also has up-to-date caches stored which can be used new PRs as fallback making CI runs faster even for the initial run.

File: .github/workflows/reusable-verification.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      - name: Setup Cloud Formation Linter with Latest Version
+        uses: scottbrenner/cfn-lint-action@v2
+
+      - name: Run linters for Cloudformation
+        run: |
+          make lint-cfn
+
+      - name: Run linters for Helm
+        run: |
+          make lint-cfn
```
Comment (chrisgacsal) [inline .github/workflows/reusable-verification.yml:57]: `s/make lint-cfn/make lint-helm/`

File: Makefile [hunk 2]
```diff
@@ -256,5 +260,5 @@ bin/actionlint-$(ACTIONLINT_VERSION): | $(BIN_DIR)
 	@mv bin/actionlint $@
 
 .PHONY: actionlint-lint
```
Comment (Tehsmash) [inline Makefile:262]: need to update this .PHONY to match the change below

Comment (chrisgacsal) [reply Makefile]: Nice catch! Fixed.

File: .github/workflows/reusable-verification.yml (hunk 1 above)
Comment (chrisgacsal) [reply .github/workflows/reusable-verification.yml]: Fixed.

**Review by Tehsmash — APPROVED** [review]

**Pull request merged**